compiler/shared_ast/types.ml

```diff
@@ -146,10 +154,12 @@ type ('a, 't) marked_gexpr = (('a, 't) gexpr, 't) Marked.t
     library, based on higher-order abstract syntax *)
 and ('a, 't) gexpr =
   (* Constructors common to all ASTs *)
```

Comment:
Why is `gexpr` in `types.ml` and not in `expr.ml`?

Reply:
Ah actuallu `types.ml` is not _about_ Catala types at all, it's just a module _containing_ type definitions. It's a common pattern to define one module with most type definitions, it allows to skip a redundant ml or mli file (since type defs have to be duplicated between ml and mli), and is a more reasonable use of `open` to access types but without polluting the namespace for values.

I agree that in this context the naming confusion is quite unfortunate. Maybe call the module `typedefs`, or `definitions`, or just `t` ? (It's included anyway at the toplevel of `Shared_ast` so technically you shouldn't have to use its name. Except for the quirks on error messages, see about `--short-paths`)

Reply:
`Definitions` would be better.